src/mod/endpoints/mod_sofia/sofia_reg.c

```diff
@@ -453,17 +454,25 @@ void sofia_reg_check_gateway(sofia_profile_t *profile, time_t now)
 			switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_NOTICE, "Registering %s\n", gateway_ptr->name);
 
 			if (now) {
+				if (gateway_ptr->ib_vars) {
+					custom_headers = sofia_glue_get_extra_headers_from_event(gateway_ptr->ib_vars, SOFIA_SIP_REGISTER_HEADER_PREFIX);
+				}
+
 				nua_register(gateway_ptr->nh,
 							 NUTAG_URL(gateway_ptr->register_url),
 							 TAG_IF(gateway_ptr->register_sticky_proxy, NUTAG_PROXY(gateway_ptr->register_sticky_proxy)),
 							 TAG_IF(user_via, SIPTAG_VIA_STR(user_via)),
+							 TAG_IF(custom_headers, SIPTAG_HEADER_STR(custom_headers)),
 							 SIPTAG_TO_STR(gateway_ptr->distinct_to ? gateway_ptr->register_to : gateway_ptr->register_from),
 							 SIPTAG_CONTACT_STR(gateway_ptr->register_contact),
 							 SIPTAG_FROM_STR(gateway_ptr->register_from),
 							 SIPTAG_EXPIRES_STR(gateway_ptr->expires_str),
 							 NUTAG_REGISTRAR(gateway_ptr->register_proxy),
 							 NUTAG_OUTBOUND("no-options-keepalive"), NUTAG_OUTBOUND("no-validate"), NUTAG_KEEPALIVE(0), TAG_NULL());
 				gateway_ptr->retry = now + gateway_ptr->retry_seconds;
+
+				switch_safe_free(custom_headers);
+				custom_headers = NULL;
```

Comment:
Just minor but switch_safe_free is a macro that check nullity, free it then set the variable to null. No need to set the custom_headers to NULL here